src/App.js

```diff
@@ -22,34 +22,58 @@ export function App () {
         FILE_VIEW: 1,
     };
 
-    const urlSearchParams = new URLSearchParams(window.location.search, "?");
-
     const [fileInfo, setFileInfo] = useState(null);
     const [theme, setTheme] = useState(THEME_STATES.DARK);
     const [appMode, setAppMode] = useState(null);
+    const [prettify, setPrettify] = useState(null);
+    const [logEventIdx, setLogEventIdx] = useState(null);
 
     const switchTheme = (theme) => {
         localStorage.setItem("ui-theme", theme);
         document.getElementById("app").setAttribute("data-theme", theme);
         setTheme(theme);
     };
 
-    useEffect(() => {
-        console.debug("Version:", config.version);
-        const lsTheme = localStorage.getItem("ui-theme");
-        switchTheme(lsTheme === THEME_STATES.LIGHT?THEME_STATES.LIGHT:THEME_STATES.DARK);
+    /**
+     * Loads the initial state of the viewer from url arguments.
+     *
+     * ### Example:
+     * localhost:3010/?filePath=/logs/sample.clp.zst&prettify=true&logEventIdx=1
+     *
+     * @return {[string,boolean,string]} [filePath,prettify,logEventIdx]
+     */
+    const getInitialStateFormUrl = () => {
+        const urlSearchParams = new URLSearchParams(window.location.search, "?");
```

Comment:
[URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) doesn't take two arguments. Did you mean to use VerbatimURLParams?

Reply:
I imported VerbatimURLParams from the viewer. I will eventually replace this with component form our yscope-web-common repo so there is no dependency on the viewer component in App.js.